src/core/device.c

```diff
@@ -665,13 +665,13 @@ static void device_found_changed(Device *d, DeviceFound previous, DeviceFound no
 }
 
 static void device_update_found_one(Device *d, DeviceFound found, DeviceFound mask) {
-        Manager *m;
+        Manager *m = NULL;
 
         assert(d);
 
         m = UNIT(d)->manager;
 
-        if (MANAGER_IS_RUNNING(m) && (m->honor_device_enumeration || MANAGER_IS_USER(m))) {
+        if (m != NULL && MANAGER_IS_RUNNING(m) && (m->honor_device_enumeration || MANAGER_IS_USER(m))) {
```

Comment:
Is this solving some actual issue, e.g. a crash? The manager should be always set for all units, since they can hardly exist when the manager is running.

Reply:
@keszybz thanks for the review. 

m = UNIT(d)->manager; => This can set m to NULL

Reply:
Please read what I wrote above: unit objects are created as children of the manager object.